internal/cmd/plugin/color.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright The CloudNativePG Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/logrusorgru/aurora/v4"
+	"github.com/spf13/cobra"
+	"golang.org/x/term"
+)
+
+// colorConfiguration represents how the output should be colorized.
+// It is a `pflag.Value`, therefore implements String(), Set(), Type()
+type colorConfiguration string
+
+const (
+	// colorAlways configures the output to always be colorized
+	colorAlways colorConfiguration = "always"
+	// colorAuto configures the the output to be colorized only when attached to a terminal
+	colorAuto colorConfiguration = "auto"
+	// colorNever configures the the output never to be colorized
+	colorNever colorConfiguration = "never"
+)
+
+// String returns the strig representation
+func (e colorConfiguration) String() string {
```

Comment:
I think this should be a pointer, right? to have the requested value of `e` updated

Reply:
checked the invoker is only read the value, probably string(colorConfiguration) also works